src/rust/ide/lib/view/src/graph_editor/component/breadcrumbs.rs

```diff
@@ -0,0 +1,524 @@
+        }
+
+
+        // === Debug ===
+
+        frp::extend! { network
+
+            // === Push ===
+
+            indices <= frp.debug.push_breadcrumb.map(f!((local_call)
+                model.debug_push_breadcrumb(local_call)
+            ));
+            old_breadcrumb <- indices.map(f!((indices) model.get_breadcrumb(indices.0)));
+            new_breadcrumb <- indices.map(f!((indices) model.get_breadcrumb(indices.1)));
+            eval old_breadcrumb([] (breadcrumb) breadcrumb.as_ref().map(|breadcrumb| {
+                breadcrumb.frp.deselect.emit(());
+            }));
+            eval new_breadcrumb([] (breadcrumb) breadcrumb.as_ref().map(|breadcrumb| {
+                breadcrumb.frp.select.emit(());
+                breadcrumb.frp.fade_in.emit(());
+            }));
+
+
+            // === Pop ===
+
+            indices <= frp.debug.pop_breadcrumb.map(f!((_) model.debug_pop_breadcrumb()));
+            old_breadcrumb <- indices.map(f!((indices) model.get_breadcrumb(indices.0)));
+            new_breadcrumb <- indices.map(f!((indices) model.get_breadcrumb(indices.1)));
+            eval old_breadcrumb([] (breadcrumb) breadcrumb.as_ref().map(|breadcrumb| {
+                breadcrumb.frp.deselect.emit(());
+            }));
+            eval new_breadcrumb([] (breadcrumb) breadcrumb.as_ref().map(|breadcrumb| {
+                breadcrumb.frp.select.emit(());
+            }));
+
+
+            // === Select ===
+            selected <- frp.debug.select_breadcrumb.map(f!((index)
+                model.debug_select_breadcrumb(*index))
+            );
+
+            popped_count <= selected.map(f!([](selected)
+                (0..selected.0).collect_vec()
+            ));
+            local_calls  <= selected.map(f!([](selected)
+                selected.1.clone()
+            ));
+            eval popped_count((_) frp.debug.pop_breadcrumb.emit(()));
+            eval local_calls((local_call)
+                frp.debug.push_breadcrumb.emit(local_call)
+            );
+
+        }
+
+
+        // === Project Name ===
+
+        frp::extend! { network
+            eval frp.project_name((name) model.project_name.frp.name.emit(name));
+            frp.outputs.project_name <+ model.project_name.frp.outputs.name;
+        }
+
+
+        // === GUI Update ===
+
+        frp::extend! { network
+            eval model.project_name.frp.outputs.width((width) {
+                model.relayout_for_project_name_width(*width)
+            });
+        }
+
+
+        // === User Interaction ===
+
+        frp::extend! {network
+            _mouse_down <- model.project_name.frp.outputs.mouse_down.map(f!([frp,model] (_) {
+                let (popped_count,local_calls) = model.select_breadcrumb(0);
+                for _ in 0..popped_count {
+                    frp.outputs.breadcrumb_pop.emit(());
+                }
+                for local_call in local_calls {
+                    frp.outputs.breadcrumb_push.emit(local_call);
+                }
```

Comment:
thisshould be distributed to frp network by using the `<=` operator. So, for each `0..popped_count` you should emit empty event `()` and connect it to popping utility. Basically, `map` blocks should be very small. Emiting more than 1 event from `map` block should be considered an anti pattern (I haven't seen a situation where it really is needed)